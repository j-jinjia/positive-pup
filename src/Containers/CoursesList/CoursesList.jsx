import CourseCard from "../../components/CourseCard/CourseCard"
import "./CoursesList.scss"

export const CoursesList = ({courseData}) => {
    const cardsListJSX = courseData.map((card) => {
        return <CourseCard key={card.courseHeading} image={card.image} courseType={card.courseType} courseHeading={card.courseHeading} labelsArray={card.labelsArray} />
    })

    return (
    <div className="courses-list">{cardsListJSX}</div>
  )
}
