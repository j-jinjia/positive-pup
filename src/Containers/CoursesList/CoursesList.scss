.courses-list {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-wrap: wrap;
  gap: 2rem;
}
