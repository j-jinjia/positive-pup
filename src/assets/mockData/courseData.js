import image0 from "../mockData/images/image.png";
import image1 from "../mockData/images/image1.png";
import image2 from "../mockData/images/image2.png";
import image3 from "../mockData/images/image3.png";

const courses = [
  {
    image: image0,
    courseTypeSingular: "Group Class",
    courseTypePlural: "Group Classes",
    courseHeading: "Little Stars Course",
    labelsArray: ["In Person", "10 weeks+"],
  },
  {
    image: image1,
    courseTypeSingular: "Group Class",
    courseTypePlural: "Group Classes",
    courseHeading: "Rising Stars Course",
    labelsArray: ["In Person", "16 weeks+"],
  },
  {
    image: image2,
    courseTypeSingular: "Bespoke Package",
    courseTypePlural: "Bespoke Packages",
    courseHeading: "1:1 Personalised Training",
    labelsArray: ["In Person", "Intensive"],
  },
  {
    image: image0,
    courseTypeSingular: "Group Class",
    courseTypePlural: "Group Classes",
    courseHeading: "Little Stars Course",
    labelsArray: ["Online"],
  },

  {
    image: image1,
    courseTypeSingular: "Online Course",
    courseTypePlural: "Online Courses",
    courseHeading: "Chaos to Calm",
    labelsArray: ["Online", "Flexible"],
  },

  {
    image: image3,
    courseTypeSingular: "Online Course",
    courseTypePlural: "Online Courses",
    courseHeading: "Rapid Recall",
    labelsArray: ["Online"],
  },
];

export default courses;
