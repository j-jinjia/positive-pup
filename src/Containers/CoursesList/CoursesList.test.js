import { customRender } from "../../utils/testUtils";
import { CoursesList } from "./CoursesList";
import courseData from "../../assets/mockData/courseData";

it("should match CoursesList component snapshot", () => {
  const { container } = customRender(<CoursesList courseData={courseData} />);

  expect(container).toMatchSnapshot();
});
