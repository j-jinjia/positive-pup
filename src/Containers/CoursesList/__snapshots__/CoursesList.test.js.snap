// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match CoursesList component snapshot 1`] = `
<div>
  <div
    class="courses-list"
  >
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Group Class
        </p>
        <p
          class="course-card__course-heading"
        >
          Little Stars Course
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              In Person
            </span>
          </div>
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              10 weeks+
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image1.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Group Class
        </p>
        <p
          class="course-card__course-heading"
        >
          Rising Stars Course
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              In Person
            </span>
          </div>
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              16 weeks+
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image2.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Bespoke Package
        </p>
        <p
          class="course-card__course-heading"
        >
          1:1 Personalised Training
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              In Person
            </span>
          </div>
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              Intensive
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Group Class
        </p>
        <p
          class="course-card__course-heading"
        >
          Little Stars Course
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              Online
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image1.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Online Course
        </p>
        <p
          class="course-card__course-heading"
        >
          Chaos to Calm
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              Online
            </span>
          </div>
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              Flexible
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image3.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Online Course
        </p>
        <p
          class="course-card__course-heading"
        >
          Rapid Recall
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              Online
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Bespoke Package
        </p>
        <p
          class="course-card__course-heading"
        >
          Premium Puppy
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              Online
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="course-card"
    >
      <div
        class="course-card__img-container"
      >
        <img
          alt="dog"
          class="course-card__img"
          src="image3.png"
        />
      </div>
      <div
        class="course-card__course-content"
      >
        <p
          class="course-card__course-type"
        >
          Bespoke Package
        </p>
        <p
          class="course-card__course-heading"
        >
          Bespoke Packages
        </p>
        <div
          class="course-card__labels"
        >
          <div
            class="card-label"
          >
            <span
              class="card-label__text"
            >
              Online
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
